### PR TITLE
feat: run snapshot updates as tests

### PR DESCRIPTION
**Pull request by jbedard**
### Type of change

- Bug fix (change which fixes an issue)
- Refactor (a code change that neither fixes a bug or adds a new feature)

**For changes visible to end-users**

- Breaking change (this change will force users to change their own code or config)
- Relevant documentation has been updated

### Test plan

- Covered by existing test cases
- Manual testing; please provide instructions so we can reproduce:

### CODE REVIEW

File: jest/defs.bzl [hunk 1]
```diff
@@ -54,7 +47,7 @@ def jest_test(
         auto_configure_reporters = True,
         auto_configure_test_sequencer = True,
         snapshots_ext = ".snap",
-        quiet_snapshot_updates = False,
+        quiet_snapshot_updates = False,  # TODO: no longer supported
```
Comment (jbedard) [inline jest/defs.bzl:50]: Need to either delete this or implement it another way.

File: jest/private/jest_test.bzl [hunk 2]
```diff
@@ -98,6 +98,9 @@ def _impl(ctx):
         # useful if the user configures jest-junit themselves.
         fixed_env["JEST_JUNIT_OUTPUT_FILE"] = "$$XML_OUTPUT_FILE"
 
+    if ctx.attr.quiet_snapshot_updates:
+        fixed_env["JS_BINARY__SILENT_ON_SUCCESS"] = "1"
```
Comment (jbedard) [inline jest/private/jest_test.bzl:102]: Is using `JS_BINARY__SILENT_ON_SUCCESS` like `js_run_binary` ok or is that considered "private"?

Comment (gregmagolan) [reply jest/private/jest_test.bzl:102]: That one I think we can consider public as it is very useful and not likely to every change.

Could even add it to the docs of js_binary

File: jest/private/bazel_snapshot_resolver.cjs [hunk 3]
```diff
@@ -45,53 +50,42 @@ let snapshotResolver = process.env.JEST_TEST__USER_SNAPSHOT_RESOLVER
...
+snapshotResolver = {
+  ...origSnapshotResolver,
+
+  resolveSnapshotPath: (testPath) => {
+    let orig = origSnapshotResolver.resolveSnapshotPath(testPath);
+
+    if (path.isAbsolute(orig)) {
+      orig = path.join(
+        process.env.BUILD_WORKSPACE_DIRECTORY,
+        path.relative(process.cwd(), orig)
```
Comment (gregmagolan) [inline jest/private/bazel_snapshot_resolver.cjs:70]: hmmm. this is probably going to be a problem if there is a `chdir`. I think you need a test case to handle that.

Comment (jbedard) [reply jest/private/bazel_snapshot_resolver.cjs]: Fixed, see https://github.com/aspect-build/rules_jest/compare/4dd40f91d5213de63146d500b2ae73700c7833ac..a9d43790eba6ba1a768281332ab49879408eddaa

**Review by gregmagolan — APPROVED** [review]

**Comment by Aghassi** [comment]
Tested this internally with our stuff. This works in the cases we spoke about:

1. no snapshot folder exists, but a snapshot is expected to be written. Bazel will write the generated snapshot during the run back to the source tree
2. there is a top level snapshot folder in a given folder and then a nested folder within that (e.g. `src` and `src/utils`) that requires a snapshot folder. Both folders will get snapshot folders

**Pull request merged**